src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java

```diff
@@ -223,6 +224,10 @@ protected synchronized AlgorithmParameters engineGetParameters() {
                 final AlgorithmParameters params = AlgorithmParameters.getInstance("RSASSA-PSS");
                 params.init(pssParams_);
                 return params;
+            } catch (final NoSuchAlgorithmException ex) {
+                // NOTE: this method can only throw unchecked exceptions, and UnsupportedOperationException
+                // seems like the most appropriate for JDK platforms that don't support RSASSA-PSS.
+                throw new UnsupportedOperationException(ex.getMessage());
```

Comment:
By doing this we lose the stack trace. Please be sure to include the caught exception as the cause in the thrown exception.
```suggestion
                throw new UnsupportedOperationException(ex);
```

Reply:
Sorry, this was a hold-over from when I was using InvalidParameterException, which [doesn't have a constructor that takes a `Throwable`](https://docs.oracle.com/javase/7/docs/api/java/security/InvalidParameterException.html). I've adjusted this line to wrap the exception and add a brief message.